app/controllers/lists_controller.rb

```diff
@@ -0,0 +1,34 @@
+class ListsController < ApplicationController
+  before_action :set_list, only: [:destroy]
+  before_action :set_board, only: [:create]
+
+  def destroy
+    @board = @list.board
+    @list.destroy
+    flash[:notice] = "The list and its tasks were deleted"
+    redirect_to @board
+  end
+
+  def create
+    @list = List.new(list_params)
+    if @list.save
+      redirect_to @board
+    else
+      flash[:danger] = "Something went wrong." 
+      redirect_to @board
+    end
+  end
+
+  private 
+  def set_board
+    @board = Board.find_by_id(list_params[:board_id])
+  end
+
+  def set_list
+    @list = List.find_by_id(params[:id])
+  end
```

Comment:
there is a syntactic sugar out there we can use in these cases. 

it goes like this

```rb
def board
  @board ||= Board.find_by......
end
```

whenever you need to call the board, you can just use the board method. 

if it wasn't defined (it's the first time running this time), it will be declared; else, it is already stored in memory.